### PR TITLE
Change new diff name & Fix editor crashing the game because of deleted difficulties

### PR DESCRIPTION
**Pull request by Kizuna-dev**
Change the name of a newly created difficulty to it's creation date only, this fix the map not uploading correctly because of special character.

When deleting a difficulty from a mapset, then try to edit a difficulty from said mapset, the game will crash. This is because of the game is trying to open a non existant file. To prevent this, i added exception handler to prevent the editor from reading those file. I also made it so the game automatically delete the map from the cache, and also reload the database* so the difficulty get removed when going back in the selection screen.

*The reload haven't been tested on a large database, i don't know how much it can impact the editor loading.

(i'm also not sure if this is the best way to handle this situation, but it work.)

### CODE REVIEW

File: Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs [hunk 1]
```diff
@@ -61,7 +62,17 @@ public EditorAutoModPanelContainer(EditScreen screen)
             {
                 if (x == Screen.Map)
                     return;
-
+                try
+                {
+                    File.OpenRead(Directory.GetCurrentDirectory() + "\\Songs\\" + x.Directory + "\\" + x.Path);
```
Comment (YaLTeR) [inline Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs:67]: \ is windows-specific, please use /

File: Quaver.Shared/Screens/Edit/UI/Menu/EditorFileMenuBar.cs [hunk 2]
```diff
@@ -347,6 +348,14 @@ private void CreateViewSection()
 
                 foreach (var map in Screen.Map.Mapset.Maps)
                 {
+                    try
+                    {
+                        File.OpenRead(Directory.GetCurrentDirectory() + "\\Songs\\" + map.Directory + "\\" + map.Path);
```
Comment (YaLTeR) [inline Quaver.Shared/Screens/Edit/UI/Menu/EditorFileMenuBar.cs:353]: same

File: Quaver.Shared/Screens/Edit/UI/Menu/EditorFileMenuBar.cs [hunk 3]
```diff
@@ -347,6 +348,14 @@ private void CreateViewSection()
 
                 foreach (var map in Screen.Map.Mapset.Maps)
                 {
+                    try
+                    {
+                        File.OpenRead(Directory.GetCurrentDirectory() + "/Songs/" + map.Directory + "/" + map.Path);
+                    } catch (FileNotFoundException)
+                    {
+                        break;
+                    }
```
Comment (Swan) [inline Quaver.Shared/Screens/Edit/UI/Menu/EditorFileMenuBar.cs:357]: This isn't something you're going to want to do every single frame. If the goal is to fix the crash when attempting to load the file, then that should be handled when the user clicks it (and display a notification).

File: Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs [hunk 4]
```diff
@@ -61,7 +62,17 @@ public EditorAutoModPanelContainer(EditScreen screen)
...
-
+                try
+                {
+                    File.OpenRead(Directory.GetCurrentDirectory() + "/Songs/" + x.Directory + "/" + x.Path);
+                } catch (FileNotFoundException)
+                {
+                    mapset.Remove(x.LoadQua());
+                    MapDatabaseCache.RemoveMap(x);
+                    MapDatabaseCache.Load(false);
+                    return;
```
Comment (Swan) [inline Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs:73]: I don't think the AutoMod should be responsible for removing files/updating the cache. If files aren't able to be read, then the AutoMod should notify the user of the issue.

File: Quaver.Shared/Database/Maps/Map.cs [hunk 5]
```diff
@@ -383,6 +383,16 @@ public virtual Qua LoadQua(bool checkValidity = false)
...
                     var quaPath = $"{ConfigManager.SongDirectory}/{Directory}/{Path}";
+
+                    try
+                    {
+                        File.OpenText(quaPath);
+                    } catch (FileNotFoundException)
+                    {
+                        qua = new Qua();
+                        break;
+                    }
```
Comment (Swan) [inline Quaver.Shared/Database/Maps/Map.cs:394]: Very odd way of handling this. If anything, this error handling should be done when parsing in `Qua.cs`

**Review by Swan — CHANGES_REQUESTED** [review]

File: Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs (hunk 4 above)
Comment (Kizuna-dev) [reply Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs]: Moved the database update in the EditorScreen file.

File: Quaver.Shared/Screens/Edit/UI/Menu/EditorFileMenuBar.cs (hunk 3 above)
Comment (Kizuna-dev) [reply Quaver.Shared/Screens/Edit/UI/Menu/EditorFileMenuBar.cs]: Found another way and fixed with https://github.com/Quaver/Quaver.API/pull/117

File: Quaver.Shared/Database/Maps/Map.cs (hunk 5 above)
Comment (Kizuna-dev) [reply Quaver.Shared/Database/Maps/Map.cs]: Done with https://github.com/Quaver/Quaver.API/pull/117

**Pull request closed without merging**